app/services/permission_service.rb

```diff
@@ -8,12 +8,22 @@ def initialize(user)
   end
 
   def allow?(controller, action)
-    @controller = controller
-    @action     = action
     if user.platform_admin?
+      platform_admin_permissions(controller, action)
+    elsif user.store_admin?
+      store_admin_permissions(controller, action)
+    elsif user.registered_user?
+      registered_user_permissions(controller, action)
+    else
+      guest_permissions(controller, action)
+    end
+  end
+
+  private
+    def platform_admin_permissions(controller, action)
       return true if controller == 'home'            && action == 'home'
       return true if controller == 'sessions'        && action.in?(%w(new create destroy))
-      return true if controller == 'cars'            && action.in?(%w(index show))
+      return true if controller == 'cars'            && action.in?(%w(index show destroy))
```

Comment:
Line is too long. [90/80]<br>Operator `&&` should be surrounded with a single space.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.